wpilib/wpilib/pidcontroller.py

```diff
@@ -519,41 +507,91 @@ def reset(self):
             self.totalError = 0
             self.result = 0
 
+    def removeListeners(self):
+        if self.pEntry is not None:
+            self.pEntry.removeListener(self.pListener)
```

Comment:
The `None` exception is occurring because you aren't setting `self.pEntry` to None (though I'm not 100% sure why this is).

Reply:
> if it's not none
> code fails because it's none
🤔

Reply:
I figured it out, the global resources weren't getting cleared out, so free/reset were being called multiple times because the object never went away.

However, it's probably a good idea to set all the entries/etc to None when removing them, so let's do that.